lib/ads/ad_utils.js

```diff
@@ -40,11 +40,7 @@ shaka.ads.Utils = class {
       if (!creatives) {
         continue;
       }
-      for (const creative of TXml.findChildren(creatives, 'Creative')) {
-        const linear = TXml.findChild(creative, 'Linear');
-        if (!linear) {
-          continue;
-        }
+      const processLinearAd = (linear) => {
```

Comment:
This method (and the new `processLinearAd` method below) are both long enough that I feel like it might be better to make them into private static methods on the `shaka.ads.Utils` class. Taking in `startTime` and `interstitials` as additional arguments. Locally-defined functions like these make debugging a little harder.

Reply:
Done!